transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -504,15 +505,142 @@ def _handle_event(self, event):
         else:
             self._logger.warn('Unhandled BLE event: ' + str(event))
 
+
     def _parse_scan_response(self, response):
         """Parse the IOTile specific data structures in the BLE advertisement packets and add the device to our list of scanned devices
         """
+        self._logger.debug("   call: _parse_scan_response({0})".format(response))
```

Comment:
These debug logging statements seem like they're leftover from initial development learning what functions were calling what and should be removed before.  In particular:

- these statements seem one or two levels two chatty.  This one could be called hundred of times per second if there is a single non-iotile ble device around that advertises frequently
- logging statements should not use `.format` to format the log message.  I know that generally `.format` is preferred for string replacement but the `logging` module is an exception to that rule even in python 3.  The reason is that if you call .format then the string replacement happens immediately every time `debug` is called which is potentially expensive, even if the user hasn't enabled that level of logging.  Just passing the arguments to `log` instead allows the logging module to see if log statement is sent anywhere before doing any argument conversion.  See:
https://docs.python.org/3/library/logging.html#logging.Logger.debug